paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -317,28 +318,30 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
         // before adding the Fragment.
         val paymentMethodsValue = paymentMethods.value
 
-        return if (
-            stripeIntentValue != null &&
+        return stripeIntentValue != null &&
             paymentMethodsValue != null &&
             isGooglePayReadyValue != null &&
             isResourceRepositoryReadyValue != null &&
             isLinkReadyValue != null &&
             savedSelectionValue != null
-        ) {
-            FragmentConfig(
-                stripeIntent = stripeIntentValue,
-                isGooglePayReady = isGooglePayReadyValue,
-                savedSelection = savedSelectionValue
-            )
-        } else {
-            null
-        }
     }
 
-    fun transitionTo(target: TransitionTargetType) {
+    abstract fun transitionToFirstScreen()
+
+    protected fun transitionTo(target: TransitionTarget) {
         _transition.postValue(Event(target))
     }
 
+    fun transitionToAddPaymentScreen() {
+        transitionTo(TransitionTarget.AddAnotherPaymentMethod)
+    }
+
+    internal sealed class TransitionTarget {
+        object SavedPaymentMethods : TransitionTarget()
```

Comment:
nit: Given that the other two start with verbs:

```suggestion
        object SelectSavedPaymentMethods : TransitionTarget()
```